go-interface-tutorial/main_test.go

```diff
@@ -2,33 +2,97 @@
 package main
 
 import (
+	"database/sql"
+	"log"
+	"strings"
 	"testing"
-	"time"
 )
 
 type MockShopDB struct{}
 
-func (m *MockShopDB) CountCustomers(_ time.Time) (int, error) {
-	return 1000, nil
-}
+// func (m *MockShopDB) CountCustomers(_ time.Time) (int, error) {
+// 	return 1000, nil
+// }
+
+// func (m *MockShopDB) CountSales(_ time.Time) (int, error) {
+// 	return 333, nil
+// }
 
-func (m *MockShopDB) CountSales(_ time.Time) (int, error) {
-	return 333, nil
+func (m *MockShopDB) CreateBooks() (bool, error) {
+	return true, nil
 }
 
-func TestCalculateSalesRate(t *testing.T) {
+func TestCreateBooks(t *testing.T) {
 	// Initialize the mock.
 	m := &MockShopDB{}
 	// Pass the mock to the calculateSalesRate() function.
-	sr, err := calculateSalesRate(m)
+	sr, err := createBooks(m)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	// Check that the return value is as expected, based on the mocked
 	// inputs.
-	exp := "0.33"
+	exp := "true"
 	if sr != exp {
 		t.Fatalf("got %v; expected %v", sr, exp)
 	}
 }
+
+// func TestCalculateSalesRate(t *testing.T) {
+// 	// Initialize the mock.
+// 	m := &MockShopDB{}
+// 	// Pass the mock to the calculateSalesRate() function.
+// 	sr, err := calculateSalesRate(m)
+// 	if err != nil {
+// 		t.Fatal(err)
+// 	}
+
+// 	// Check that the return value is as expected, based on the mocked
+// 	// inputs.
+// 	exp := "0.33"
+// 	if sr != exp {
+// 		t.Fatalf("got %v; expected %v", sr, exp)
+// 	}
+// }
+
+func TestPostgresQueries(t *testing.T) {
+	connStr := "host=localhost port=5400 user=docker password=docker sslmode=disable"
+	db, err := sql.Open("postgres", connStr)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+	p := PostgresRepository{
+		sdb: &ShopDB{db},
+	}
+	addBookWithTitle("Cutie")
```

Comment:
Can you extract out the `"Cutie"` string to a variable and use that instead everywhere?